### PR TITLE
Configuration parse fix

### PR DESCRIPTION
**Pull request by mjarco**

### CODE REVIEW

**Comment by medzin** [comment]
@mjarco Any chance for some CI build for this?

File: config/config.go [hunk 1]
```diff
@@ -102,21 +120,16 @@ func Configure() (conf Config, err error) {
...
-		}
-		conf.BackendURLs = append(conf.BackendURLs, url)
-	}
-
-	if len(conf.MaintainedBackend) > 0 {
-		u, perr := url.Parse(conf.MaintainedBackend)
-		if perr != nil {
-			return conf, perr
+	confFile, openErr := os.Open(*confFilePath)
+	if openErr == nil {
```
Comment (medzin) [inline config/config.go]: Invert `if` statement to reduce nesting.

File: httphandler/httphandler.go [hunk 2]
```diff
@@ -89,10 +90,13 @@ func NewHandler(conf config.Config) http.Handler {
 		Dial:                dialer.Dial,
 		DisableKeepAlives:   conf.KeepAlive,
 		MaxIdleConnsPerHost: int(conf.ConnLimit)}
-
+	backends := make([]*url.URL, len(conf.Backends))
+	for _, backend := range(conf.Backends) {
+		backends = append(backends, backend.URL)
```
Comment (medzin) [inline httphandler/httphandler.go]: Why appending? You already have a slice with required length, so you can just assign values to indexes.

File: config/config.go [hunk 3]
```diff
@@ -36,19 +36,40 @@ type YamlConfig struct {
...
+
+
+func (j *YAMLURL) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var s string
+	if err := unmarshal(&s); err != nil {
+		return err
+	}
+	url, err := url.Parse(s)
+	if url.Host == "" {
+		return fmt.Errorf("url should match proto://host[:port]/path scheme, got %q", s)
```
Comment (medzin) [inline config/config.go]: How this message will look when the `s` variable is empty?

File: config/config.go [hunk 4]
```diff
@@ -102,21 +120,16 @@ func Configure() (conf Config, err error) {
...
+	confFile, openErr := os.Open(*confFilePath)
+	if openErr == nil {
+		yconf, parseErr := parseConf(confFile)
+		if parseErr != nil {
+			return conf, parseErr
 		}
-		conf.MaintainedBackend = u.Host
+		conf = Config{YamlConfig: yconf}
+	} else {
+		fmt.Println("Cannot read config file:", openErr.Error())
```
Comment (mjarco) [inline config/config.go]: I'll remove above print. I would rather use log.Fatal, but as all loggers log to syslog error message would not be visible to user. So I'll leave fmt.Print in main.go and return in main func

File: config/config.go (hunk 3 above)
Comment (mjarco) [inline config/config.go]: It will never run this function in that case.

Comment (medzin) [inline config/config.go]: Can you write test for case when Host will be empty?

File: config/config_test.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import (
+	"github.com/go-yaml/yaml"
+	"testing"
+)
+
+type TestYaml struct {
+	Field YAMLURL
+}
+
+func TestYAMLURLParsing(t *testing.T) {
```
Comment (medzin) [inline config/config_test.go]: Split to two functions with more meaningful test names. Something like `TestIfParsesCorrectYAML`, `TestIfParsesIncorrectYAML`.

File: config/config_test.go [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+	if err != nil {
+		t.Error(err.Error())
+	}
+}
+
+func TestYAMLURLParsingFailure(t *testing.T) {
+	incorrect := []byte(`field: golang.org:80/pkg/net`)
+	testyaml := TestYaml{}
+	err := yaml.Unmarshal(incorrect, &testyaml)
+	if err == nil {
```
Comment (medzin) [inline config/config_test.go]: `assert.Error(err)`

File: config/config_test.go [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+func TestYAMLURLParsingEmpty(t *testing.T) {
+	incorrect := []byte(`field:`)
+	testyaml := TestYaml{}
+	err := yaml.Unmarshal(incorrect, &testyaml)
+	if err != nil {
+		t.Errorf("Should not even try to parse")
+	}
+	if testyaml.Field.URL != nil {
```
Comment (medzin) [inline config/config_test.go]: `assert.Nil()`

File: config/config.go [hunk 8]
```diff
@@ -36,19 +36,37 @@ type YamlConfig struct {
 	//List request methods to be logged in synclog in case of backend failure
 	SyncLogMethods []string `yaml:"SyncLogMethods,omitempty"`
 	//Should we keep alive connections with backend servers
-	KeepAlive bool `yaml:"SyncLogMethods,omitempty"`
+	KeepAlive bool `yaml:"KeepAlive,omitempty"`
```
Comment (wendigo) [inline config/config.go]: I guess that omitempty does not make sense here - zero value of bool is false

File: config/config_test.go [hunk 9]
```diff
@@ -14,28 +15,20 @@ func TestYAMLURLParsingSuccessful(t *testing.T) {
...
 	incorrect := []byte(`field:`)
 	testyaml := TestYaml{}
 	err := yaml.Unmarshal(incorrect, &testyaml)
-	if err != nil {
-		t.Errorf("Should not even try to parse")
-	}
-	if testyaml.Field.URL != nil {
-		t.Errorf("Should be nil")
-	}
+	assert.Nil(t, err, "Should not even try to parse")
```
Comment (medzin) [inline config/config_test.go]: `assert.NoError`

File: config/config_test.go [hunk 10]
```diff
@@ -14,28 +15,20 @@ func TestYAMLURLParsingSuccessful(t *testing.T) {
 	correct := []byte(`field: http://golang.org:80/pkg/net`)
 	testyaml := TestYaml{}
 	err := yaml.Unmarshal(correct, &testyaml)
-	if err != nil {
-		t.Error(err.Error())
-	}
+	assert.Nil(t, err, "Should be correct")
```
Comment (medzin) [inline config/config_test.go]: `assert.Error`

**Pull request merged**